### PR TITLE
Add KBIC65 keyboard

### PR DESCRIPTION
**Pull request by b-karl**
## Description

After receiving requests to add QMK and VIA support for the [KBIC65 keyboard](https://github.com/b-karl/KBIC65) I would like to officially merge my keyboard definition with the QMK master branch. It is an open-source 65% FR4-stack keyboard with through-hole diodes and a ProMicro footprint, for all features please see [the GitHub repo for the board](https://github.com/b-karl/KBIC65) itself.
 
## Types of Changes

- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by b-karl** [comment]
Items I believe are left:
- [ ] Harmonize keymaps between default and VIA
- [ ] Create default ANSI and ISO layouts
- [ ] Create an alternative full-key layout
- [ ] Test firmware again on real board

File: keyboards/karlb/kbic65/info.json [hunk 1]
```diff
@@ -0,0 +1,402 @@
+{
+    "keyboard_name": "KBIC65",
+    "url": "https://karlb.eu/kbic65/",
+    "maintainer": "b-karl",
+    "width": 16.5,
+    "height": 5.25,
```
Comment (drashna) [inline keyboards/karlb/kbic65/info.json:6]:
```suggestion
```

File: keyboards/karlb/kbic65/info.json [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+            "D0",
+            "E6",
+            "D4",
+            "D7",
+            "C6",
+            "D2"
+        ]
+    },
+    "usb": {
+        "vid": "0xFEED",
```
Comment (drashna) [inline keyboards/karlb/kbic65/info.json:32]:
```suggestion
        "vid": "0x424B",
```

File: keyboards/karlb/kbic65/rules.mk [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/karlb/kbic65/rules.mk:10]: We do recommend enabling bootmagic lite by default.  Especially, as it's enabled as part of VIA.
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/karlb/kbic65/rules.mk [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/karlb/kbic65/rules.mk:18]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by b-karl** [comment]
@drashna Thanks for the comments/changes, have accepted all of them. Looked around and compared to other merged keyboard PRs, but just to double check, is there any other documentation then the readme.md for the keyboard itself that I should update? And are there any tests I should implement or run, except for testing the final firmware on an actual keyboard?

File: keyboards/karlb/kbic65/config.h [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+#define PRODUCT         kbic65
+
+/* key matrix size */
+#define MATRIX_ROWS 9
+#define MATRIX_COLS 8
+#define MATRIX_ROW_PINS { B5, D1,  B4, D0, E6, D4, D7, C6, D2 }
+#define MATRIX_COL_PINS { B2, B6, B3, B1, F7, F6, F5, F4 }
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION ROW2COL
```
Comment (drashna) [inline keyboards/karlb/kbic65/config.h:38]: This is redundant. 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by b-karl** [comment]
Sorry for keeping this sitting, busy life. Will try to work on it soon to finish the last points that I commented on previously.

**Review by drashna — APPROVED** [review]

File: keyboards/karlb/kbic65/rules.mk [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/karlb/kbic65/rules.mk:15]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

File: keyboards/karlb/kbic65/rules.mk [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
```
Comment (fauxpark) [inline keyboards/karlb/kbic65/rules.mk:14]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks

**Pull request merged**